drivers/accel/adxl314/adxl314.c

```diff
@@ -0,0 +1,1120 @@
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param size         - The number of bytes to be read and returned in read_data.
+ * @param read_data    - The read data buffer.
+ *
+ * @return ret         - Result of the reading operation (0 - success, negative
+ *                       value for failure).
+*******************************************************************************/
+int adxl314_read_device_data(struct adxl314_dev *dev, uint8_t base_address,
+			     uint16_t size, uint8_t *read_data)
+{
+	int ret;
+
+	if (dev->comm_type == ADXL314_SPI_COMM) {
+		dev->comm_buff[0] = ADXL314_SPI_READ | base_address;
+		if (size > 1)
+			dev->comm_buff[0] |= ADXL314_MULTIBIT;
+		ret = no_os_spi_write_and_read(dev->com_desc.spi_desc, dev->comm_buff,
+					       1 + size);
+		for (uint16_t idx = 0; idx < size; idx++)
+			read_data[idx] = dev->comm_buff[idx+1];
+	} else {
+		ret = no_os_i2c_write(dev->com_desc.i2c_desc, &base_address, 1, 0);
+		if (ret)
+			return ret;
+		ret = no_os_i2c_read(dev->com_desc.i2c_desc, read_data, size, 1);
+	}
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Write data to the device.
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param size         - The number of bytes to be written. It is the size
+ *                       of the write_data buffer.
+ * @param write_data   - The data which is going to be written.
+ *
+ * @return ret         - Result of the writing operation (0 - success, negative
+ *                       value for failure).
+*******************************************************************************/
+int adxl314_write_device_data(struct adxl314_dev *dev, uint8_t base_address,
+			      uint16_t size, uint8_t *write_data)
+{
+	int ret;
+
+	for (uint16_t idx = 0; idx < size; idx++)
+		dev->comm_buff[1+idx] = write_data[idx];
+
+	if (dev->comm_type == ADXL314_SPI_COMM) {
+		dev->comm_buff[0] = ADXL314_SPI_WRITE | base_address;
+		if (size > 1)
+			dev->comm_buff[0] |= ADXL314_MULTIBIT;
+		ret = no_os_spi_write_and_read(dev->com_desc.spi_desc, dev->comm_buff,
+					       size + 1);
+	} else {
+		dev->comm_buff[0] = base_address;
+		ret = no_os_i2c_write(dev->com_desc.i2c_desc, dev->comm_buff, size + 1, 1);
+	}
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Set or clear a bit of a specific register.
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param action       - Set/Reset bit action.
+ * @param modified_bit - The bit that is set/reset.
+ *
+ * @return ret - Result of the operation (0 - success, negative value - failure).
+*******************************************************************************/
+int adxl314_control_reg_bit(struct adxl314_dev *dev,
```

Comment:
Can this be accomplished as well with `adxl314_reg_write_msk()`.

Reply:
Yes, it can. I will remove the adxl314_control_reg_bit function.